### PR TITLE
renew type checking

### PR DESCRIPTION
**Pull request by linjing-lab**
- change "pyplot" to plt(matplotlib.pyplot)
- change Optional[str, int] to Optional[Union[str, int]]
- add Optional to type checking

### CODE REVIEW

File: paddlets/analysis/analysis_report.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ def __init__(
         self._columns = columns
         self._analyzers = self._get_analyzers(names, params)
 
-    def export_docx_report(self, path: str = ".", file_name: str = "analysis_report.docx") -> None:
+    def export_docx_report(self, path: Optional[str] = ".", file_name: Optional[str] = "analysis_report.docx") -> None:
```
Comment (kehuo) [inline paddlets/analysis/analysis_report.py:72]: should NOT be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/analysis/analysis_report.py [hunk 2]
```diff
@@ -139,7 +139,7 @@ def export_docx_report(self, path: str = ".", file_name: str = "analysis_report.
         document.save(path)
         logger.info(f"save report succcess, save at {path}")
     
-    def export_json_report(self, log: bool = True) -> Dict:
+    def export_json_report(self, log: Optional[bool] = True) -> Dict:
```
Comment (kehuo) [inline paddlets/analysis/analysis_report.py:142]: should NOT be None, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/analysis/analysis_report.py [hunk 3]
```diff
@@ -233,7 +233,7 @@ def _get_analyzers(self, names: Union[str, List[str]], params: Dict = None) -> L
 
         return analyzers
 
-    def get_all_analyzers_names(self, log: bool = True) -> List[str]:
+    def get_all_analyzers_names(self, log: Optional[bool] = True) -> List[str]:
```
Comment (kehuo) [inline paddlets/analysis/analysis_report.py:236]: should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/analysis/frequency_domain.py [hunk 4]
```diff
@@ -43,7 +43,7 @@ class FFT(Analyzer):
         None
     """
 
-    def __init__(self, fs: float = 0, norm: bool = True, half: bool = True, **kwargs):
+    def __init__(self, fs: Optional[float] = 0, norm: Optional[bool] = True, half: Optional[bool] = True, **kwargs):
```
Comment (kehuo) [inline paddlets/analysis/frequency_domain.py:46]: should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/analysis/frequency_domain.py [hunk 5]
```diff
@@ -225,16 +225,16 @@ class STFT(Analyzer):
...
-        window: Union[str, Tuple[str], List[str]] = 'hann',
-        nperseg: int = 256,
-        noverlap: Union[None, int] = None,
-        nfft: Union[None, int] = None,
-        detrend: Union[str, bool] = False,
-        return_onesided: bool = True,
-        boundary: Union[str, None] = 'zeros',
-        padded: bool = True,
-        axis: int = -1,
+        fs: Optional[float] = 1.0,
```
Comment (kehuo) [inline paddlets/analysis/frequency_domain.py:228]: should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/analysis/frequency_domain.py [hunk 6]
```diff
@@ -225,16 +225,16 @@ class STFT(Analyzer):
...
+        window: Optional[Union[str, Tuple[str], List[str]]] = 'hann',
```
Comment (kehuo) [inline paddlets/analysis/frequency_domain.py:229]: should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/analysis/frequency_domain.py [hunk 7]
```diff
@@ -225,16 +225,16 @@ class STFT(Analyzer):
...
+        nperseg: Optional[int] = 256,
```
Comment (kehuo) [inline paddlets/analysis/frequency_domain.py:230]: should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/analysis/frequency_domain.py [hunk 8]
```diff
@@ -225,16 +225,16 @@ class STFT(Analyzer):
...
+        noverlap: Optional[Union[None, int]] = None,
+        nfft: Optional[Union[None, int]] = None,
+        detrend: Optional[Union[str, bool]] = False,
```
Comment (kehuo) [inline paddlets/analysis/frequency_domain.py:233]: should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/analysis/frequency_domain.py [hunk 9]
```diff
@@ -225,16 +225,16 @@ class STFT(Analyzer):
...
+        return_onesided: Optional[bool] = True,
```
Comment (kehuo) [inline paddlets/analysis/frequency_domain.py:234]: should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/analysis/frequency_domain.py [hunk 10]
```diff
@@ -225,16 +225,16 @@ class STFT(Analyzer):
...
+        boundary: Optional[Union[str, None]] = 'zeros',
```
Comment (kehuo) [inline paddlets/analysis/frequency_domain.py:235]: should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/analysis/frequency_domain.py [hunk 11]
```diff
@@ -225,16 +225,16 @@ class STFT(Analyzer):
...
+        padded: Optional[bool] = True,
```
Comment (kehuo) [inline paddlets/analysis/frequency_domain.py:236]: should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/analysis/frequency_domain.py [hunk 12]
```diff
@@ -225,16 +225,16 @@ class STFT(Analyzer):
...
+        axis: Optional[int] = -1,
```
Comment (kehuo) [inline paddlets/analysis/frequency_domain.py:237]: should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/analysis/time_domain.py [hunk 13]
```diff
@@ -36,10 +36,10 @@ class Seasonality(Analyzer):
...
     def __init__(self, 
                  period: Optional[int] = None,
-                 nlags: int = 300,
-                 alpha: float = 0.05,
-                 mode: str = 'additive',
-                 order: int = 1,
+                 nlags: Optional[int] = 300,
+                 alpha: Optional[float] = 0.05,
+                 mode: Optional[str] = 'additive',
+                 order: Optional[int] = 1,
```
Comment (kehuo) [inline paddlets/analysis/time_domain.py:42]: Line39 - Line42 should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/analysis/time_domain.py [hunk 14]
```diff
@@ -84,7 +84,7 @@ def analyze(
     def _seasonality(
         self, 
         X: pd.DataFrame,
-        period_dict: dict = {},
+        period_dict: Optional[dict] = {},
```
Comment (kehuo) [inline paddlets/analysis/time_domain.py:87]: should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/analysis/time_domain.py [hunk 15]
```diff
@@ -364,8 +364,8 @@ class Correlation(Analyzer):
 
     """
     def __init__(self, 
-                 method: str = 'pearson',
-                 lag: int = 0,
+                 method: Optional[str] = 'pearson',
```
Comment (kehuo) [inline paddlets/analysis/time_domain.py:367]: should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/analysis/time_domain.py [hunk 16]
```diff
@@ -364,8 +364,8 @@ class Correlation(Analyzer):
...
+                 lag: Optional[int] = 0,
```
Comment (kehuo) [inline paddlets/analysis/time_domain.py:368]: should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/automl/autots.py [hunk 17]
```diff
@@ -92,20 +92,20 @@ def __init__(
...
+            k_fold: Optional[Union[str, int]] = 'auto',
+            metric: Optional[str] = 'auto',
+            mode: Optional[str] = 'auto',
+            refit: Optional[bool] = True,
+            ensemble: Optional[bool] = False,
             local_dir: Optional[str] = None,
-            n_jobs: int = -1,
-            verbose: int = 4
+            n_jobs: Optional[int] = -1,
+            verbose: Optional[int] = 4
```
Comment (kehuo) [inline paddlets/automl/autots.py:108]: Line95 - Line105 and Line107 - Line108 should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/automl/autots.py [hunk 18]
```diff
@@ -179,10 +179,10 @@ def __init__(
     def fit(
             self,
             train_tsdataset: Union[TSDataset, List[TSDataset]],
-            valid_tsdataset: Union[TSDataset, List[TSDataset]] = None,
-            n_trials: int = 20,
-            cpu_resource: float = 1.0,
-            gpu_resource: float = 0
+            valid_tsdataset: Optional[Union[TSDataset, List[TSDataset]]] = None,
+            n_trials: Optional[int] = 20,
+            cpu_resource: Optional[float] = 1.0,
```
Comment (kehuo) [inline paddlets/automl/autots.py:184]: should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/automl/autots.py [hunk 19]
```diff
@@ -179,10 +179,10 @@ def __init__(
...
+            gpu_resource: Optional[float] = 0
```
Comment (kehuo) [inline paddlets/automl/autots.py:185]: should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/automl/optimize_runner.py [hunk 20]
```diff
@@ -109,18 +109,18 @@ def optimize(self,
                  in_chunk_len: int,
                  out_chunk_len: int,
                  train_tsdataset: Union[TSDataset, List[TSDataset]],
-                 valid_tsdataset: Union[TSDataset, List[TSDataset]] = None,
-                 sampling_stride: int = 1,
-                 skip_chunk_len: int = 0,
+                 valid_tsdataset: Optional[Union[TSDataset, List[TSDataset]]] = None,
+                 sampling_stride: Optional[int] = 1,
+                 skip_chunk_len: Optional[int] = 0,
```
Comment (kehuo) [inline paddlets/automl/optimize_runner.py:114]: Line113 and Line 114 should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/automl/optimize_runner.py [hunk 21]
```diff
@@ -109,18 +109,18 @@ def optimize(self,
...
-                 cpu_resource: float = 1.0,
-                 gpu_resource: float = 0,
+                 metric: Optional[ABCMeta] = MAE,
+                 mode: Optional[str] = "min",
+                 resampling_strategy: Optional[str] = "holdout",
+                 split_ratio: Optional[float] = 0.1,
+                 k_fold: Optional[int] = 3,  # cv的fold切分数, 默认5折切分
+                 n_trials: Optional[int] = 5,
+                 cpu_resource: Optional[float] = 1.0,
+                 gpu_resource: Optional[float] = 0,
```
Comment (kehuo) [inline paddlets/automl/optimize_runner.py:123]: Line 116 - Line123 should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/splitter.py [hunk 22]
```diff
@@ -29,13 +29,13 @@ class SplitterBase(metaclass=ABCMeta):
...
-                 skip_size: int = 0,
-                 verbose: bool = True) -> None:
+                 skip_size: Optional[int] = 0,
+                 verbose: Optional[bool] = True) -> None:
         self._skip_size = skip_size
         self._verbose = verbose
 
     def split(self, dataset: TSDataset,
-              return_index: bool = False) -> Union[TSDataset, pd.DatetimeIndex, pd.RangeIndex]:
+              return_index: Optional[bool] = False) -> Union[TSDataset, pd.DatetimeIndex, pd.RangeIndex]:
```
Comment (kehuo) [inline paddlets/datasets/splitter.py:38]: should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/splitter.py [hunk 23]
```diff
@@ -306,11 +306,11 @@ class ExpandingWindowSplitter(SplitterBase):
...
-    def __init__(self, n_splits: int = 5,
-                 test_size: Union[int, None] = None,
-                 skip_size: int = 0,
-                 max_train_size: int = None,
-                 verbose: bool = True) -> None:
+    def __init__(self, n_splits: Optional[int] = 5,
+                 test_size: Optional[Union[int, None]] = None,
+                 skip_size: Optional[int] = 0,
+                 max_train_size: Optional[int] = None,
+                 verbose: Optional[bool] = True) -> None:
```
Comment (kehuo) [inline paddlets/datasets/splitter.py:313]: (1) Line309 n_splits should Not be Optional.
(2) Line310 can be either Optional[int] = None, or Union[int, None] = None. The reason is that Optional[x] is equivalent to Union[x, None].
(3) Line311 skip_size  should Not be Optional.
(4) Line313 verbose should Not be Optional.
Refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/splitter.py [hunk 24]
```diff
@@ -479,9 +479,9 @@ class SlideWindowSplitter(SplitterBase):
     def __init__(self,
                  train_size: int,
                  test_size: int,
-                 step_size: int = None,
-                 skip_size: int = 0,
-                 verbose: bool = True) -> None:
+                 step_size: Optional[int] = None,
+                 skip_size: Optional[int] = 0,
```
Comment (kehuo) [inline paddlets/datasets/splitter.py:483]: Line483 skip_size should Not be Optional, refers to  https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/splitter.py [hunk 25]
```diff
@@ -479,9 +479,9 @@ class SlideWindowSplitter(SplitterBase):
...
+                 verbose: Optional[bool] = True) -> None:
```
Comment (kehuo) [inline paddlets/datasets/splitter.py:484]: Line484 verbose should Not be Optional, refers to  https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/tsdataset.py [hunk 26]
```diff
@@ -86,7 +86,7 @@ def load_from_dataframe(
         time_col: Optional[str] = None,
         value_cols: Optional[Union[List[str], str]] = None,
         freq: Optional[Union[str, int]] = None,
-        drop_tail_nan: bool = False,
+        drop_tail_nan: Optional[bool] = False,
```
Comment (kehuo) [inline paddlets/datasets/tsdataset.py:89]: Line89 drop_tail_nan should Not be Optional, refers to  https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/tsdataset.py [hunk 27]
```diff
@@ -255,7 +255,7 @@ def astype(self, dtype: Union[np.dtype, type, Dict[str, Union[np.dtype, type]]])
         self._data = self._data.astype(dtype)
         return self
 
-    def to_dataframe(self, copy: bool=True) -> pd.DataFrame:
+    def to_dataframe(self, copy: Optional[bool]=True) -> pd.DataFrame:
```
Comment (kehuo) [inline paddlets/datasets/tsdataset.py:258]: Line258 copy should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/tsdataset.py [hunk 28]
```diff
@@ -271,7 +271,7 @@ def to_dataframe(self, copy: bool=True) -> pd.DataFrame:
         else:
             return self.data
     
-    def to_numpy(self, copy: bool=True) -> np.ndarray:
+    def to_numpy(self, copy: Optional[bool]=True) -> np.ndarray:
```
Comment (kehuo) [inline paddlets/datasets/tsdataset.py:274]: Line274 copy should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/tsdataset.py [hunk 29]
```diff
@@ -290,7 +290,7 @@ def to_numpy(self, copy: bool=True) -> np.ndarray:
     def get_index_at_point(
         self, 
         point: Union[pd.Timestamp, str, float, int], 
-        after=True
+        after: Optional[bool]=True
```
Comment (kehuo) [inline paddlets/datasets/tsdataset.py:293]: Line294 after should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/tsdataset.py [hunk 30]
```diff
@@ -357,7 +357,7 @@ def get_index_at_point(
     def split(
         self, 
         split_point: Union[pd.Timestamp, str, float, int], 
-        after=True
+        after: Optional[bool]=True
```
Comment (kehuo) [inline paddlets/datasets/tsdataset.py:360]: Line360 copy should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/tsdataset.py [hunk 31]
```diff
@@ -778,10 +778,10 @@ def load_from_csv(
...
         static_cov_cols: Optional[Union[List[str], str]] = None,
         freq: Optional[Union[str, int]] = None,
-        fill_missing_dates: bool = False,
-        fillna_method: str = "pre",
-        fillna_window_size: int = 10,
-        drop_tail_nan: bool = False,
+        fill_missing_dates: Optional[bool] = False,
+        fillna_method: Optional[str] = "pre",
+        fillna_window_size: Optional[int] = 10,
+        drop_tail_nan: Optional[bool] = False,
```
Comment (kehuo) [inline paddlets/datasets/tsdataset.py:784]: Line781 - Line784 should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/tsdataset.py [hunk 32]
```diff
@@ -855,10 +855,10 @@ def load_from_dataframe(
...
```
Comment (kehuo) [inline paddlets/datasets/tsdataset.py:861]: Line858 - Line861 should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/tsdataset.py [hunk 33]
```diff
@@ -1482,7 +1482,7 @@ def set_column(
         self,
         column: Union[str, int],
         value: Union[pd.Series, str, int],
-        type: str = 'known_cov'
+        type: Optional[str] = 'known_cov'
```
Comment (kehuo) [inline paddlets/datasets/tsdataset.py:1485]: Line1485 should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/tsdataset.py [hunk 34]
```diff
@@ -1654,13 +1654,13 @@ def drop(
...
-             low_quantile:float = 0.05,
-             high_quantile:float = 0.95,
-             central_quantile:float = 0.5,
-             **kwargs) -> "pyplot":
+             columns: Optional[Union[List[str], str]] = None, 
+             add_data: Optional[Union[List["TSDataset"], "TSDataset"]] = None,
+             labels: Optional[Union[List[str], str]] = None,
+             low_quantile: Optional[float] = 0.05,
+             high_quantile: Optional[float] = 0.95,
+             central_quantile: Optional[float] = 0.5,
```
Comment (kehuo) [inline paddlets/datasets/tsdataset.py:1662]: Line1660 - Line1662 should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/tsdataset.py [hunk 35]
```diff
@@ -1800,9 +1800,9 @@ def plot(self,
         return plot
 
     def _fill_between_quantiles(self,
-                                column: str = None,
-                                low_quantile: float = 0.05,
-                                high_quantile: float = 0.95,
+                                column: Optional[str] = None,
+                                low_quantile: Optional[float] = 0.05,
```
Comment (kehuo) [inline paddlets/datasets/tsdataset.py:1804]: Line1804 should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/tsdataset.py [hunk 36]
```diff
@@ -1800,9 +1800,9 @@ def plot(self,
...
+                                high_quantile: Optional[float] = 0.95,
```
Comment (kehuo) [inline paddlets/datasets/tsdataset.py:1805]: Line1805 should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/tsdataset.py [hunk 37]
```diff
@@ -1963,7 +1963,7 @@ def freq(self):
         return self._freq
 
     @classmethod
-    def concat(cls, tss: List["TSDataset"], axis: int = 0, drop_duplicates = True, keep = 'first') -> "TSDataset":
+    def concat(cls, tss: List["TSDataset"], axis: Optional[int] = 0, drop_duplicates: Optional[bool]=True, keep: Optional[str]='first') -> "TSDataset":
```
Comment (kehuo) [inline paddlets/datasets/tsdataset.py:1966]: Line1966 axis,  drop_duplicates, keep should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/datasets/tsdataset.py [hunk 38]
```diff
@@ -2071,7 +2071,7 @@ def dtypes(self) -> pd.Series:
             type_list.append(pd.DataFrame(self._static_cov, index=[0]).dtypes)
         return pd.concat(type_list)
 
-    def sort_columns(self, ascending: bool = True):
+    def sort_columns(self, ascending: Optional[bool] = True):
```
Comment (kehuo) [inline paddlets/datasets/tsdataset.py:2074]: Line2074 ascending should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/ensemble/ensemble_forecaster_base.py [hunk 39]
```diff
@@ -32,9 +32,9 @@ class EnsembleForecasterBase(BaseModel, metaclass=abc.ABCMeta):
     def __init__(self,
                  in_chunk_len: int,
                  out_chunk_len: int,
-                 skip_chunk_len: int = 0,
-                 estimators: List[Tuple[object, dict]] = None,
-                 verbose=False
+                 skip_chunk_len: Optional[int] = 0,
```
Comment (kehuo) [inline paddlets/ensemble/ensemble_forecaster_base.py:35]: Line35 skip_chunk_len should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/ensemble/ensemble_forecaster_base.py [hunk 40]
```diff
@@ -152,7 +152,7 @@ def _predict_estimators(self,
             predictions.append(estimator.predict(tsdataset))
         return predictions
 
-    def save(self, path: str, ensemble_file_name: str = "paddlets-ensemble-partial.pkl") -> None:
+    def save(self, path: str, ensemble_file_name: Optional[str] = "paddlets-ensemble-partial.pkl") -> None:
```
Comment (kehuo) [inline paddlets/ensemble/ensemble_forecaster_base.py:155]: Line155 ensemble_file_name should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/ensemble/ensemble_forecaster_base.py [hunk 41]
```diff
@@ -186,7 +186,7 @@ def save(self, path: str, ensemble_file_name: str = "paddlets-ensemble-partial.p
         self._estimators = model_tmp
 
     @staticmethod
-    def load(path: str, ensemble_file_name: str = "paddlets-ensemble-partial.pkl") -> "EnsembleForecasterBase":
+    def load(path: str, ensemble_file_name: Optional[str] = "paddlets-ensemble-partial.pkl") -> "EnsembleForecasterBase":
```
Comment (kehuo) [inline paddlets/ensemble/ensemble_forecaster_base.py:189]: Line189 ensemble_file_name should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/ensemble/stacking_ensemble_forecaster.py [hunk 42]
```diff
@@ -40,12 +40,12 @@ def __init__(self,
...
-                 resampling_strategy: str = 'cv',
-                 split_ratio: Union[str, float] = 0.1,
-                 k_fold: Union[str, int] = 3,
-                 verbose: bool = False
+                 final_learner: Optional[Callable] = None,
+                 use_backtest: Optional[bool] = True,
+                 resampling_strategy: Optional[str] = 'cv',
+                 split_ratio: Optional[Union[str, float]] = 0.1,
+                 k_fold: Optional[Union[str, int]] = 3,
+                 verbose: Optional[bool] = False
```
Comment (kehuo) [inline paddlets/ensemble/stacking_ensemble_forecaster.py:48]: Line44 - Line48  use_backtest, resampling_strategy, split_ratio, split_ratio, k_fold, verbose should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

File: paddlets/ensemble/weighting_ensemble_forecaster.py [hunk 43]
```diff
@@ -37,8 +37,8 @@ def __init__(self,
                  out_chunk_len: int,
                  skip_chunk_len: int,
                  estimators: List[Tuple[object, dict]],
-                 mode="mean",
-                 verbose: bool = False
+                 mode: Optional[str]="mean",
+                 verbose: Optional[bool] = False
```
Comment (kehuo) [inline paddlets/ensemble/weighting_ensemble_forecaster.py:41]: Line40 - Line41 mode, verbose should Not be Optional, refers to https://docs.python.org/3/library/typing.html#typing.Optional

**Review by kehuo — CHANGES_REQUESTED** [review]
Thanks for the commit. As according to [python typing.Optional doc](https://docs.python.org/3/library/typing.html#typing.Optional), only if an explicit value of None is allowed for an argument, the use of typing.Optional is appropriate, so please remove all inappropriate Optional type checking.

**Comment by linjing-lab** [comment]
> Thanks for the commit. As according to [python typing.Optional doc](https://docs.python.org/3/library/typing.html#typing.Optional), only if an explicit value of None is allowed for an argument, the use of typing.Optional is appropriate, so please remove all inappropriate Optional type checking.

ok

**Pull request closed without merging**